### ISSUE TITLE
error: ‘gpiod_line_close_chip’ was not declared in this scope

### ISSUE THREAD

**Issue by RabAusten** [aquaticus/nexus433#27]
I'm getting the error:
******************************************************
pi@pi4build:~/Projects/debug $ make

Scanning dependencies of target version
[  0%] Built target version
Scanning dependencies of target nexus433
[ 12%] Building CXX object CMakeFiles/nexus433.dir/config.cpp.o
[ 25%] Building CXX object CMakeFiles/nexus433.dir/decoder.cpp.o
[ 37%] Building CXX object CMakeFiles/nexus433.dir/led.cpp.o
[ 50%] Building CXX object CMakeFiles/nexus433.dir/main.cpp.o
/home/pi/Projects/nexus433/main.cpp: In function ‘int main(int, char**)’:
/home/pi/Projects/nexus433/main.cpp:348:9: error: ‘gpiod_line_close_chip’ was not declared in this scope
         gpiod_line_close_chip(line);
         ^~~~~~~~~~~~~~~~~~~~~
/home/pi/Projects/nexus433/main.cpp:348:9: note: suggested alternative: ‘gpiod_line_get_chip’
         gpiod_line_close_chip(line);
         ^~~~~~~~~~~~~~~~~~~~~
         gpiod_line_get_chip
/home/pi/Projects/nexus433/main.cpp:360:9: error: ‘gpiod_line_close_chip’ was not declared in this scope
         gpiod_line_close_chip(line);
         ^~~~~~~~~~~~~~~~~~~~~
/home/pi/Projects/nexus433/main.cpp:360:9: note: suggested alternative: ‘gpiod_line_get_chip’
         gpiod_line_close_chip(line);
         ^~~~~~~~~~~~~~~~~~~~~
         gpiod_line_get_chip
/home/pi/Projects/nexus433/main.cpp:378:5: error: ‘gpiod_line_close_chip’ was not declared in this scope
     gpiod_line_close_chip(line);
     ^~~~~~~~~~~~~~~~~~~~~
/home/pi/Projects/nexus433/main.cpp:378:5: note: suggested alternative: ‘gpiod_line_get_chip’
     gpiod_line_close_chip(line);
     ^~~~~~~~~~~~~~~~~~~~~
     gpiod_line_get_chip
make[2]: *** [CMakeFiles/nexus433.dir/build.make:102: CMakeFiles/nexus433.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/nexus433.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
*****************************************************

I followed the installation steps given in the Readme, but I'm lost as to whether it is something I have done wrong.

This is being built on a Raspberry Pi 4 B+ running the latest 32 Bit Pi OS with full desktop

Thanks
James

**Comment by aquaticus** [comment]
Have you installed `libgpiod`?

**Comment by RabAusten** [comment]
> Have you installed `libgpiod`?

Yes, and I didn't get any errors in that part. I'll try cleaning down and reinstalling from the top. I've clearly missed something.

**Comment by pmmphd** [comment]
I get the exact same error on build. I have a Raspberry Pi 3 Model B Rev 1.2 running Raspbian GNU/Linux 10 (buster) Lite. As far as I can tell, **libgpiod**  installed without errors.

**Comment by pmmphd** [comment]
It looks like **gpiod_line_close_chip()** has been removed from `libgpiod`. See [here](https://kernel.googlesource.com/pub/scm/libs/libgpiod/libgpiod/+/8cfa3e3f2f7a37e4336e815610bfdb6eefe35e8c). The fix shown in Issue #21 works in this case, too:

Install libgpiod C library (version 1.4.2 seems fine for the 4.19 kernel on Raspbian buster)
```
git clone git://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git
cd libgpiod
git checkout v1.4.2 -b v1.4.2
./autogen.sh --enable-tools=yes
make
sudo make install
```

_Originally posted by @thorgrin in https://github.com/aquaticus/nexus433/issues/21#issuecomment-596073776_

**Comment by RabAusten** [comment]
> It looks like **gpiod_line_close_chip()** has been removed from `libgpiod` (see [here](https://kernel.googlesource.com/pub/scm/libs/libgpiod/libgpiod/+/8cfa3e3f2f7a37e4336e815610bfdb6eefe35e8c)). The fix shown in Issue #21 works in this case, too:
> 
> Install libgpiod C library (version 1.4.2 seems fine for the 4.19 kernel on Raspbian buster)
> 
> ```
> git clone git://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git
> cd libgpiod
> git checkout v1.4.2 -b v1.4.2
> ./autogen.sh --enable-tools=yes
> make
> sudo make install
> ```
> 
> _Originally posted by @thorgrin in [#21 (comment)](https://github.com/aquaticus/nexus433/issues/21#issuecomment-596073776)_

Thanks, as I never got the #21 error, I didn't run that. Will do so and report back.

**Comment by aquaticus** [comment]
Problem fixed in commit 8b26319
Removed call to  gpiod_line_close_chip().
